backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/service/SpecificationDatabaseService.kt

```diff
@@ -22,29 +22,24 @@ class SpecificationDatabaseService constructor(
 ) : SpecificationPersistenceService {
 
     override fun save(specification: Specification) {
-        val specificationEntity = specificationEntityMapper.fromDomain(specification)
+        val newEntity = specificationEntityMapper.fromDomain(specification)
 
-        if (specificationRepository.existsByTitle(specificationEntity.title)) {
-            val existingEntity = specificationRepository.findByTitle(specificationEntity.title)
+        val entityToStore = specificationRepository.findById(newEntity.id)
+            .map { existingEntity -> existingEntity.pureUpdate(newEntity)}
+            .orElse(newEntity)
 
-            val versions = existingEntity.versions + specificationEntity.versions
-            val newEntity = SpecificationEntity(existingEntity.id, existingEntity.title, existingEntity.description, versions, existingEntity.remoteAddress)
-
-            mapAndStoreEntity(newEntity)
-        } else {
-            mapAndStoreEntity(specificationEntity)
-        }
+        mapAndStoreEntity(entityToStore)
     }
 
     override fun findAll(): List<Specification> = specificationRepository.findAll().map { spec -> specificationEntityMapper.toDomain(spec) }
 
-    override fun findOne(id: UUID): Specification? =
-        specificationRepository.findById(id).orElse(null)?.let { spec -> specificationEntityMapper.toDomain(spec) }
+    override fun findOne(id: ServiceId): Specification? =
+        specificationRepository.findById(id.id).orElse(null)?.let { spec -> specificationEntityMapper.toDomain(spec) }
 
-    override fun delete(id: UUID) = specificationRepository.deleteById(id)
+    override fun delete(id: ServiceId) = specificationRepository.deleteById(id.id)
```

Comment:
Unfortunately not, that was my first attempt. It's because `@Id` annotated fields must be primitive types. However I just found out [here](https://www.baeldung.com/spring-mvc-custom-data-binder) that (in response to a different comment of yours) I can automatically convert Spring's `@PathVariables` to `ServiceId`s